### PR TITLE
[clang] Make deprecations of some `FileManager` APIs formal

### PR DESCRIPTION
**Pull request by jansvoboda11**
Some `FileManager` APIs still return `{File,Directory}Entry` instead of the preferred `{File,Directory}EntryRef`. These are documented to be deprecated, but don't have the attribute that warns on their usage. This PR marks them as such with `LLVM_DEPRECATED()` and replaces their usage with the recommended counterparts. NFCI.

### CODE REVIEW

File: clang-tools-extra/clang-move/tool/ClangMove.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ int main(int argc, const char **argv) {
       for (auto I = Files.begin(), E = Files.end(); I != E; ++I) {
         OS << "  {\n";
         OS << "    \"FilePath\": \"" << *I << "\",\n";
-        const auto Entry = FileMgr.getFile(*I);
+        const auto Entry = FileMgr.getOptionalFileRef(*I);
```
Comment (benlangmuir) [inline clang-tools-extra/clang-move/tool/ClangMove.cpp:202]: Having `*Entry` on the next line without checking isn't great, but obviously not a new issue in this PR you need to fix.

File: clang/lib/Frontend/CompilerInstance.cpp [hunk 2]
```diff
@@ -427,12 +427,8 @@ static void InitializeFileRemapping(DiagnosticsEngine &Diags,
     }
 
     // Create the file entry for the file that we're mapping from.
-    const FileEntry *FromFile =
-        FileMgr.getVirtualFile(RF.first, ToFile->getSize(), 0);
-    if (!FromFile) {
-      Diags.Report(diag::err_fe_remap_missing_from_file) << RF.first;
```
Comment (benlangmuir) [inline clang/lib/Frontend/CompilerInstance.cpp:433]: Nice catch

File: clang/lib/Frontend/CompilerInstance.cpp [hunk 3]
```diff
@@ -1925,8 +1921,8 @@ ModuleLoadResult CompilerInstance::findOrCompileModuleAndReadAST(
     M = HS.lookupModule(ModuleName, ImportLoc, true, !IsInclusionDirective);
 
     // Check whether M refers to the file in the prebuilt module path.
-    if (M && M->getASTFile())
-      if (auto ModuleFile = FileMgr->getFile(ModuleFilename))
+    if (M && M->getASTFile() && M->getASTFile())
```
Comment (benlangmuir) [inline clang/lib/Frontend/CompilerInstance.cpp:1924]: Redundant condition

Comment (jansvoboda11) [reply clang/lib/Frontend/CompilerInstance.cpp:1924]: Good catch. I was playing around with squashing this into a single `if` statement like so:

```c++
    // Check whether M refers to the file in the prebuilt module path.
    if (M && M->getASTFile() &&
        *M->getASTFile() == FileMgr->getOptionalFileRef(ModuleFilename))
      return M;
```

but didn't like how it wraps.

**Review by benlangmuir — APPROVED** [review]
Thanks for cleaning this up!

**Pull request merged**